### PR TITLE
[OC-1441] New monitoring screen

### PR DESCRIPTION
**Pull request by vlo-rte**
Release notes : 

In Features section : 

[OC-1441](https://opfab.atlassian.net/browse/OC-1441) : New monitoring screen

### CODE REVIEW

File: ui/main/src/app/model/line-of-monitoring-result.model.ts [hunk 1]
```diff
@@ -17,8 +18,8 @@ export interface LineOfMonitoringResult {
     title: I18n;
     summary: I18n;
     processName: string;
-    coordinationStatus: string;
-    coordinationStatusColor: string;
     cardId: string;
     severity: string;
+    processId: string;
+    processState: TypeOfStateEnum
```
Comment (freddidierRTE) [inline ui/main/src/app/model/line-of-monitoring-result.model.ts:24]: rename processState to typeOfState

File: ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.html [hunk 2]
```diff
@@ -15,21 +15,23 @@
             <th scope="col"></th>
             <th scope="col" style="padding-left: 5px" translate>monitoring.time</th>
             <th scope="col" translate>monitoring.businessPeriod</th>
+            <th scope="col" translate *ngIf="displayServiceColumn">monitoring.filters.service</th>
             <th scope="col" translate>monitoring.filters.process</th>
             <th scope="col" translate>monitoring.title</th>
             <th scope="col" translate>monitoring.summary</th>
-            <th scope="col" translate>monitoring.status</th> 
+            <th scope="col" translate>monitoring.processState</th>
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.html:22]: processState -- > processTypeOfState

File: ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.html [hunk 3]
```diff
@@ -15,21 +15,23 @@
...
         <tr *ngFor="let line of result" (click)="selectCard(line.cardId)" class="opfab-monitoring-table-line" style="cursor: pointer;">
             <td class="opfab-monitoring-sev opfab-monitoring-sev-{{line.severity}}"></td>
             <td style="padding-left: 5px">{{displayTime(line.creationDateTime)}}</td>
             <td>{{displayTime(line.beginningOfBusinessPeriod)}} - {{displayTime(line.endOfBusinessPeriod)}}</td>
+            <td translate *ngIf="displayServiceColumn">{{ processesService.findServiceLabelForProcess(line.processId) }}</td>
             <td translate>{{line.processName}}</td>
             <td translate [translateParams]="line.title.parameters">{{line.title.key}}</td>
             <td translate [translateParams]="line.summary.parameters">{{line.summary.key}}</td>
-            <td translate [ngStyle]="{'color': line.coordinationStatusColor}">{{line.coordinationStatus}} </td>
+            <td translate>monitoring.filters.processState.{{line.processState}} </td>
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.html:34]: processState -- > processTypeOfState

File: ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.ts [hunk 4]
```diff
@@ -60,30 +63,36 @@ export class MonitoringTableComponent implements OnDestroy {
 
         const timeColumnName = this.translateColumn('monitoring.time');
         const businessPeriodColumnName = this.translateColumn('monitoring.businessPeriod');
+        const serviceColumnName = this.translateColumn('monitoring.filters.service');
         const processColumnName = this.translateColumn('monitoring.filters.process');
         const titleColumnName = this.translateColumn('monitoring.title');
         const summaryColumnName = this.translateColumn('monitoring.summary');
-        const statusColumnName = this.translateColumn('monitoring.status');
+        const processStateColumnName = this.translateColumn('monitoring.processState');
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.ts:70]: processState -- > processTypeOfState

File: ui/main/src/app/modules/monitoring/monitoring.component.ts [hunk 5]
```diff
@@ -63,8 +63,7 @@ export class MonitoringComponent implements OnInit, OnDestroy {
                         return null;
                     }
                     return cards.map(card => {
-                            let color = 'white';
-                            let name: string;
+                            let processState: TypeOfStateEnum;
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/monitoring/monitoring.component.ts:66]: processState -- > processTypeOfState

File: ui/main/src/app/services/filter.service.ts [hunk 6]
```diff
@@ -132,6 +132,23 @@ export class FilterService {
...
+                    return processStatesList.includes(processStateOfTheCard);
+                }
+                // permissive filter
+                return true;
+            },
+            false,
+            {processStates: null}
+        );
+    }
+
```
Comment (freddidierRTE) [inline ui/main/src/app/services/filter.service.ts:151]: initProcessStateFilter -- > initProcessTypeOfStateFilter
processState -- > processTypeOfState

File: ui/main/src/app/services/filter.service.ts [hunk 7]
```diff
@@ -156,7 +174,8 @@ export enum FilterType {
     PUBLISHDATE_FILTER,
     ACKNOWLEDGEMENT_FILTER,
     TEST_FILTER,
-    PROCESS_FILTER
+    PROCESS_FILTER,
+    PROCESSSTATE_FILTER
```
Comment (freddidierRTE) [inline ui/main/src/app/services/filter.service.ts:178]: PROCESSTYPEOFSTATE_FILTER

File: ui/main/src/app/services/processes.service.ts [hunk 8]
```diff
@@ -222,4 +222,65 @@ export class ProcessesService {
...
+        const services = Array.from(object.processesDropdownListPerServices.keys());
+        services.forEach(service => object.serviceDropdownList.push({ id: service, itemName: service }));
+    }
+
+    public findServiceLabelForProcess(processId : string) : string {
+        const serviceId = this.findServiceForProcess(processId);
+        return (!! serviceId && serviceId != '') ? serviceId : "service.defaultLabel";
+    }
+
+    public makeMapOfProcessStates() : Map<string, TypeOfStateEnum> {
```
Comment (freddidierRTE) [inline ui/main/src/app/services/processes.service.ts:277]: getMapOfProcessesStatesAndTypeOfState

File: ui/main/src/assets/i18n/fr.json [hunk 9]
```diff
@@ -164,14 +164,23 @@
...
+        "FINISHED": "TERMINÉ",
+        "CANCELED": "ANNULÉ"
+      }
     },
     "time": "DATE",
     "businessPeriod": "PERIODE METIER",
     "title": "TITRE",
     "summary": "RESUME",
-    "status": "STATUS",
+    "processState": "ÉTAT DU PROCESSUS",
```
Comment (freddidierRTE) [inline ui/main/src/assets/i18n/fr.json:183]: processState --> typeOfState

File: ui/main/src/assets/i18n/fr.json [hunk 10]
```diff
@@ -164,14 +164,23 @@
     "showFilters" : "Afficher les filtres",
     "noSelection" : "NA",
     "filters": {
+      "service" : "SERVICE",
+      "selectServiceText" : "Sélectionner un Service",
       "process": "PROCESSUS",
-      "selectProcessText": "Sélectionner un Processus"
+      "selectProcessText": "Sélectionner un Processus",
+      "processState": {
```
Comment (freddidierRTE) [inline ui/main/src/assets/i18n/fr.json:171]: processState --> typeOfState

File: ui/main/src/assets/i18n/en.json [hunk 11]
```diff
@@ -164,14 +164,23 @@
     "showFilters" : "Show Filters",
     "noSelection" : "NA",
     "filters": {
-      "process": "PROCESS",
-      "selectProcessText": "Select a Process"
+      "service" : "SERVICE",
+      "selectServiceText" : "Select a Service",
+      "process" : "PROCESS",
+      "selectProcessText" : "Select a Process",
+      "processState" : {
```
Comment (freddidierRTE) [inline ui/main/src/assets/i18n/en.json:171]: processState --> typeOfState

File: ui/main/src/app/services/processes.service.ts [hunk 12]
```diff
@@ -222,4 +222,65 @@ export class ProcessesService {
...
+        let mapOfProcessStates = new Map();
+
+        for (let process of this.processes) {
+            for (let state in process.states)
+                mapOfProcessStates.set(process.id + '.' + state, process.states[state].type);
+        }
+        return mapOfProcessStates;
+    }
```
Comment (freddidierRTE) [inline ui/main/src/app/services/processes.service.ts:285]: For all the file :  replace "DropdownList" with "List" as the method are more generic and could be used for other purposes than dropdown list

File: ui/main/src/app/services/processes.service.ts [hunk 13]
```diff
@@ -222,4 +222,65 @@ export class ProcessesService {
...
 
+    public findServiceForProcess(processId : string) : string {
+        for (let group of this.processGroups) {
+            if (group.processes.find(process => process == processId))
+                return group.id;
+        }
+        return '';
+    }
+
+    public makeProcessesDropdownListPerServices(): Map<any, any> {
```
Comment (freddidierRTE) [inline ui/main/src/app/services/processes.service.ts:233]: make --> get

File: ui/main/src/app/services/processes.service.ts [hunk 14]
```diff
@@ -222,4 +222,65 @@ export class ProcessesService {
...
+            if (service != '') {
+                let processes = (!! processesDropdownListPerServices.get(service) ? processesDropdownListPerServices.get(service) : []);
+                processes.push({id: process.id, itemName: process.name, i18nPrefix: `${process.id}.${process.version}`});
+                processesDropdownListPerServices.set(service, processes);
+            }
+        });
+        return processesDropdownListPerServices;
+    }
+
+    public makeProcessesWithoutServiceDropdownList(): any[] {
```
Comment (freddidierRTE) [inline ui/main/src/app/services/processes.service.ts:248]: make -> get

File: ui/main/src/app/services/processes.service.ts [hunk 15]
```diff
@@ -222,4 +222,65 @@ export class ProcessesService {
...
+
+        this.getAllProcesses().forEach(process => {
+            const service = this.findServiceForProcess(process.id);
+            if (service == '')
+                processesWithoutServiceDropdownList.push({ id: process.id, itemName: process.name, i18nPrefix: `${process.id}.${process.version}` });
+        });
+        return processesWithoutServiceDropdownList;
+    }
+
+    public loadAllProcessesPerServices(object: {processesDropdownListPerServices: Map<any, any>;
```
Comment (freddidierRTE) [inline ui/main/src/app/services/processes.service.ts:259]: Why not doing it once on init of the service as the result will always be the same 

So I would prefer having three structure pre-calculates and 3 methods to get access : 
- getListOfServices()
- getListOfProcessPreService()
- getListOfProcessWithoutService() 

having the construction of dropDownList outside of this service as it is more related to UI consideration

**Review by freddidierRTE — COMMENTED** [review]
Add the business period + link '>>' and '<<' to navigate in time  (like it is done when timeline is hidden) 

Main points in following code review  
- Rename state in typeOfState to avoid confusion with state whenever it is needed
- I think we shoud restructure the way of building the dropdownlist as it is done inside the processService , this service shall not deal with UI stuff (Separation of concerns principal)  --> point to be discuss

File: src/test/utils/karate/businessconfig/resources/bundle_defaultProcess/config.json [hunk 16]
```diff
@@ -29,7 +30,8 @@
       "styles": [
         "style"
       ],
-      "acknowledgmentAllowed": "Always"
+      "acknowledgmentAllowed": "Always",
+      "type" : "INPROGRESS"
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply src/test/utils/karate/businessconfig/resources/bundle_defaultProcess/config.json:34]: Done

File: src/test/utils/karate/businessconfig/resources/bundle_defaultProcess/config.json [hunk 17]
```diff
@@ -39,7 +41,8 @@
...
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply src/test/utils/karate/businessconfig/resources/bundle_defaultProcess/config.json:45]: Done

File: ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.html (hunk 2 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.html]: Done. commit 1c94f451cc899441682e4aabac0218d708766314

File: ui/main/src/assets/i18n/en.json (hunk 11 above)
Comment (vlo-rte) [reply ui/main/src/assets/i18n/en.json]: Done. commit 1c94f451cc899441682e4aabac0218d708766314

File: ui/main/src/assets/i18n/fr.json (hunk 9 above)
Comment (vlo-rte) [reply ui/main/src/assets/i18n/fr.json]: Done. commit 1c94f451cc899441682e4aabac0218d708766314

File: ui/main/src/assets/i18n/fr.json (hunk 10 above)
Comment (vlo-rte) [reply ui/main/src/assets/i18n/fr.json]: Done. commit 1c94f451cc899441682e4aabac0218d708766314

File: ui/main/src/app/model/line-of-monitoring-result.model.ts (hunk 1 above)
Comment (vlo-rte) [reply ui/main/src/app/model/line-of-monitoring-result.model.ts:24]: Done.

File: ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.ts (hunk 4 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.ts]: Done.

File: ui/main/src/app/modules/monitoring/monitoring.component.ts (hunk 5 above)
Comment (vlo-rte) [reply ui/main/src/app/modules/monitoring/monitoring.component.ts:66]: Done.

File: ui/main/src/app/services/filter.service.ts (hunk 6 above)
Comment (vlo-rte) [reply ui/main/src/app/services/filter.service.ts:151]: Done.

File: ui/main/src/app/services/filter.service.ts (hunk 7 above)
Comment (vlo-rte) [reply ui/main/src/app/services/filter.service.ts:178]: Done.

File: ui/main/src/app/services/processes.service.ts (hunk 12 above)
Comment (vlo-rte) [reply ui/main/src/app/services/processes.service.ts:279]: Done. commit 79ed090dad53132056f4e5652caea5f514ed3f01

File: ui/main/src/app/services/processes.service.ts (hunk 13 above)
Comment (vlo-rte) [reply ui/main/src/app/services/processes.service.ts]: Done. commit 79ed090dad53132056f4e5652caea5f514ed3f01

File: ui/main/src/app/services/processes.service.ts (hunk 14 above)
Comment (vlo-rte) [reply ui/main/src/app/services/processes.service.ts]: Done. commit 79ed090dad53132056f4e5652caea5f514ed3f01

File: ui/main/src/app/services/processes.service.ts (hunk 8 above)
Comment (vlo-rte) [reply ui/main/src/app/services/processes.service.ts]: Done. commit 79ed090dad53132056f4e5652caea5f514ed3f01

File: ui/main/src/app/services/processes.service.ts (hunk 15 above)
Comment (vlo-rte) [reply ui/main/src/app/services/processes.service.ts]: > Why not doing it once on init of the service as the result will always be the same
> 
> So I would prefer having three structure pre-calculates and 3 methods to get access :
> 
>     * getListOfProcessGroup()
A function getProcessGroups() already exists in the file.

> 
>     * getListOfProcessPerProcessGroup()
Done : function getProcessesPerProcessGroups()
> 
>     * getListOfProcessWithoutProcessGroup()
Done : function getProcessesWithoutProcessGroup()
> 
> 
> having the construction of dropDownList outside of this service as it is more related to UI consideration

Comment (vlo-rte) [reply ui/main/src/app/services/processes.service.ts]: Done. commit 79ed090dad53132056f4e5652caea5f514ed3f01

**Comment by vlo-rte** [comment]
> Add the business period + link '>>' and '<<' to navigate in time (like it is done when timeline is hidden)
> 
> Main points in following code review
> 
>     * Rename state in typeOfState to avoid confusion with state whenever it is needed
> 
>     * I think we shoud restructure the way of building the dropdownlist as it is done inside the processService , this service shall not deal with UI stuff (Separation of concerns principal)  --> point to be discuss

Done.

File: ui/main/src/assets/styles/styles.scss [hunk 18]
```diff
@@ -834,3 +834,134 @@ html, body { height: 100%; }
...
+    }
+  }
+
+  .opfab-business-period {
+    font-size : 15px;
+    width: 100%;
+    background-color: var(--opfab-timeline-bgcolor);
+    text-align: center;
+  }
+}
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply ui/main/src/assets/styles/styles.scss]: Done. commit a061c62ca9451cd8b1e1c73bb6feeb47c64479c1

**Pull request merged**